### PR TITLE
Password recovery no email fix

### PR DESCRIPTION
**Pull request by Geesa-Vihara**
### Proposed changes in this pull request
Removed Internal server error during the password recovery when user does not have an email and added a meaningful error
Fixes - https://github.com/wso2/product-is/issues/9587

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java [hunk 1]
```diff
@@ -81,6 +81,12 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
         } catch (IdentityRecoveryException e) {
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
+            if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    throwable.getMessage())) {
+                String msg = "You have not set the users email address. Please set the email address and reset again.";
+                LOG.error(msg);
+                RecoveryUtil.handleBadRequest(msg, Constants.ERROR_CODE_EMAIL_NOT_FOUND);
```
Comment (GANGANI) [inline components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java:88]:
```suggestion
                RecoveryUtil.handleBadRequest(throwable.getMessage(), Constants.ERROR_CODE_EMAIL_NOT_FOUND);
```

**Review by GANGANI — APPROVED** [review]

File: components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java [hunk 2]
```diff
@@ -81,6 +81,11 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
         } catch (IdentityRecoveryException e) {
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
+            if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    throwable.getMessage())) {
+                LOG.error(throwable);
```
Comment (GANGANI) [inline components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java:86]:
```suggestion
                LOG.error(throwable.getMessage(), throwable);
```

File: components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java (hunk 1 above)
Comment (Geesa-Vihara) [reply components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java]: Fixed in https://github.com/Geesa-Vihara/identity-governance/commit/c153383f87b8a1de4298f5a5c9526c2dab5e297a

File: components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java (hunk 2 above)
Comment (Geesa-Vihara) [reply components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java]: Fixed in https://github.com/Geesa-Vihara/identity-governance/commit/88b15bf0fe93caf31f17c51edc8686a9941539cf

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**